test/unit/spec/common/cellComponents/tabs/jobStateListRow-Spec.js

```diff
@@ -1,78 +1,180 @@
+            expect(jobStateListRowInstance).not.toBe(null);
         });
 
         it('has the required methods', () => {
-            expect(mockJobStateListRow.start).toBeDefined();
-            expect(mockJobStateListRow.stop).toBeDefined();
+            const requiredMethods = ['start', 'stop', 'updateState', 'updateParams'];
+            requiredMethods.forEach((method) => {
+                expect(jobStateListRowInstance[method]).toBeDefined();
+            });
         });
+    });
 
-        it('should start and return the correct elements', () => {
-            mockJobStateListRow.start({
-                node: node,
-                initialState: 'created',
-                name: 'testObject',
+    describe('Starting the job state list row instance', () => {
+        let jobStateListRowInstance;
+        beforeAll(() => {
+            jobStateListRowInstance = createInstance();
+        });
+        const invalidArgs = [
+            {},
+            {
+                job: {},
+            },
+            {
+                name: {},
+                node: {},
+            },
+            {
+                name: 'this',
+                node: 'that',
+                job: undefined,
+            },
+        ];
+        invalidArgs.forEach((args) => {
+            it('will throw an error with incorrect args', () => {
+                expectAsync(jobStateListRowInstance.start(args)).toBeRejectedWithError(
+                    Error,
+                    /JobStateListRow cannot start: start argument must have the following keys:/
+                );
             });
+        });
 
-            const classContents = [
-                '.kb-job-status__cell_log_btn',
-                '.show_log',
-                '.selected_log',
-                '.kb-job-status__icon',
-                '.kb-job-status__icon--created',
-                '.kb-job-status__cell_action--created',
-            ];
-            classContents.forEach((item) => {
-                expect($(node).find(item).length).toBeGreaterThan(0);
+        JobsData.invalidJobs.forEach((invalidJob) => {
+            it('will throw an error with an invalid job', () => {
+                expectAsync(jobStateListRowInstance.start(invalidJob)).toBeRejectedWithError(
+                    Error,
+                    /JobStateListRow cannot start: invalid job object supplied/
+                );
             });
+        });
+    });
 
-            expect(node.innerHTML).toContain('CANCEL');
-            expect(node.innerHTML).toContain('Queued');
+    JobsData.validJobs.forEach((job) => {
+        beforeEach(() => {
+            jasmine.addMatchers(customMatchers);
         });
+        describe('the job state list row instance', () => {
+            describe('creates initial structure and content', () => {
+                const row = document.createElement('tr');
+                const jobStateListRowInstance = createInstance();
+                jobStateListRowInstance
```

Comment:
This should be `async / await`, also should be under an `it` instead of `describe`.